app/public/static/js/components/rotation-controls.js

```diff
@@ -0,0 +1,100 @@
+AFRAME.registerComponent("rotation-controls", {
+  schema: {
+    jsonData: {
+      parse: JSON.parse,
+      stringify: JSON.stringify,
+    },
+  },
+  multiple: true,
+  init: function () {
+    const data = this.data.jsonData;
+    let { TransformControls } = THREE;
+    let { el } = this;
+    let self = this;
+    let obj = el.object3D;
+    let sceneEl = el.sceneEl;
+    let scene = sceneEl.object3D;
+    let renderer = sceneEl.renderer;
+    let camera, controls;
+    let childId = data.id;
+
+    let childEl = document.getElementById(childId);
+    let clonedEl = document.createElement("a-gltf-model");
+    clonedEl.setAttribute("src", childEl.getAttribute("src"));
+    clonedEl.setAttribute("scale", childEl.getAttribute("scale"));
+    clonedEl.setAttribute("rotation", childEl.getAttribute("rotation"));
+
+    // TODO: Is scaling needed?
+    if (data.hasOwnProperty("position")) {
+      el.setAttribute("position", {
```

Comment:
Why are we setting the position for both `el` and `clonedEl`? What are the differences between the two?

Reply:
`el` is the outer entity with the rotation-controls. So that is the one rotating. `clonedEl` is the actual object nested inside `el`. `el` we set position because we need to.

And `clonedEl` position needs to be 0 0 0 otherwise it will start orbiting instead of rotating. Does that make sense?

Reply:
Ahhhh I see, makes sense. Is there a technical reason for having the clonedEl be separate from the outer `el`? Can we not just add the src and gltf stuff to `el` instead of `clonedEl`? Not a biggie if we can't... just wondering

Reply:
Oh truee, good point. I didnt do it originally because wasn't sure if gltf is the only stuff in there. I'll merge them into one.